modules/Bio/EnsEMBL/Hive/HivePipeline.pm

```diff
@@ -544,151 +544,203 @@ sub print_diagram {
-
+          push @{$responceStructure->{Tweaks}}, $tweakStructure;
         } elsif($tweak=~/^pipeline\.(\w+)(\?|=(.+))$/) {
+            my $tweakStructure;
             my ($attrib_name, $operator, $new_value_str) = ($1, $2, $3);
+            $tweakStructure->{Object}->{Type} = 'Pipeline';
+            $tweakStructure->{Object}->{Id} = '';
+            $tweakStructure->{Return}->{Field} = $attrib_name;
 
             if($self->can($attrib_name)) {
                 my $old_value = stringify( $self->$attrib_name() );
 
                 if($operator eq '?') {
-                    print "Tweak.Show    \tpipeline.$attrib_name ::\t$old_value\n";
+                    $tweakStructure->{Action} = 'SHOW';
+                    $tweakStructure->{Return}->{OldValue} = $old_value;
+                    $tweakStructure->{Return}->{NewValue} = $old_value;
+                    push @responce, "Tweak.Show  \tpipeline.$attrib_name ::\t$old_value\n";
                 } else {
-                    print "Tweak.Changing\tpipeline.$attrib_name ::\t$old_value --> $new_value_str\n";
+                    $tweakStructure->{Action} = 'SET';
+                    $tweakStructure->{Return}->{OldValue} = $old_value;
+                    $tweakStructure->{Return}->{NewValue} = $new_value_str;
+                    push @responce, "Tweak.Changing  \tpipeline.$attrib_name ::\t$old_value --> $new_value_str\n";
 
                     $self->$attrib_name( $new_value_str );
                     $need_write = 1;
                 }
 
             } else {
-                print "Tweak.Error   \tCould not find the pipeline-wide '$attrib_name' method\n";
+                my $error_msg = "Tweak.Error  \tCould not find the pipeline-wide '$attrib_name' method\n";
+                $tweakStructure->{Error} = $error_msg;
+                push @responce, $error_msg;
             }
-
+            push @{$responceStructure->{Tweaks}}, $tweakStructure;
         } elsif($tweak=~/^analysis\[([^\]]+)\]\.param\[(\w+)\](\?|#|=(.+))$/) {
             my ($analyses_pattern, $param_name, $operator, $new_value_str) = ($1, $2, $3, $4);
 
             my $analyses = $self->collection_of( 'Analysis' )->find_all_by_pattern( $analyses_pattern );
-            print "Tweak.Found   \t".scalar(@$analyses)." analyses matching the pattern '$analyses_pattern'\n";
+            push @responce, "Tweak.Found  \t".scalar(@$analyses)." analyses matching the pattern '$analyses_pattern'\n";
 
             my $new_value = destringify( $new_value_str );
             $new_value_str = stringify( $new_value );
 
             foreach my $analysis (@$analyses) {
-                my $analysis_name = $analysis->logic_name;
+                my $tweakStructure;
+                $tweakStructure->{Object}->{Type} = 'Analysis';
 
+                my $analysis_name = $analysis->logic_name;
                 my $old_value = $analysis->parameters;
 
+                $tweakStructure->{Object}->{Id} = $analysis->dbID . " " . $analysis_name;
```

Comment:
Since the goal is to have a parsable output, why do you merge fields ? I would suggest to put the analysis' dbID in `Id` and its logic_name in `name`

Reply:
I agree with @muffato here.

```
$tweakStructure->{Object}->{Id} = $analysis->dbID;
$tweakStructure->{Object}->{Name} = $analysis_name;
# similar for resource class
# for pipeline, can provide undef for both.
```